src/model/Board.java

```diff
@@ -89,6 +119,18 @@ public Square getSquare(int position) {
         return squares;
     }
 
+    public Set<Square> getSquaresForPlayer(PieceColor color) {
+        Set<Square> playersSquares = new HashSet<>();
```

Comment:
At that point I was comparing lots of sets of items, moves, squares, etc. And a lot of the time I really didn't care about the order from expected vs actual. Basically, when order doesn't matter, the HashSet is easier to compare and it's pretty speedy (I think?). It will never have duplicates, which could hide some programming faults (i.e. creating duplicate squares and then putting them into a set to hide the duplication), but I ensured that the code wasn't doing this beforehand. Basically I just asked myself, "Do I care about the order of this stuff?", "Do I need to access things by index?", "Do I need to hold any duplicates in my collection?", "Do I want speed and ease of comparison?". If my answers were no, no, no, yes, then sets made sense to me. You'll see a whole boat-load of sets in the MoveGenerator and in the MoveGeneratorTest classes.